co: helpers + generic interface

tomholford opened the pull request:
# Context

This PR extends `urbit-gob` by adding some devex convenience features.

# Changes

- [x] adds `ClanPoint`, `SeinPoint`, `Patp2Point`, `Point2Patp`, `Patq2Point`, and `Point2Patq`
- [x] refactors `Patp` and `Patq` function signatures to accept either `string` or `*big.Int` values
- [x] adds test coverage + helpers
- [x] update CLI  + readme to support new commands 

# Limitations

In some cases, values are converted inefficiently, e.g. from `string` --> `big.Int` --> `string`. In testing on my own hardware, this seems to introduce negligible impacts to performance, but it is worth calling out as an opportunity for future enhancement. For reference, the JS implementation apparently has a similar issue (see discussion here: https://github.com/urbit/urbit-ob/issues/18)

tomholford committed:
co: helpers + generic interface

- also added test helpers + coverage

tomholford committed:
co: more tests for patp / patq

tomholford committed:
cmd: cli supports new fns

also, update readme

tomholford commented:
@deelawn Tagging you here since I cannot add to Reviewers. PTAL at your convenience.

deelawn commented on co/co.go:
```diff
@@ -226,6 +241,33 @@ func Patq(arg string) (string, error) {
 }
 
+// Patq2Dec converts a string-encoded int or *big.Int to a @q-encoded string.
+func Patq(arg interface{}) (string, error) {
```
Can you fix the comment here?

deelawn commented on co/co.go:
```diff
@@ -211,7 +226,7 @@ func Patp2Dec(name string) (string, error) {
 
 // Patq converts a number to a @q-encoded string.
-func Patq(arg string) (string, error) {
+func patq(arg string) (string, error) {
```
I think the comment here should either be removed or `Patq` changed to `patq`

deelawn commented on co/co_test.go:
```diff
@@ -1,21 +1,56 @@
+// i2i: integer to integer
+// a2s: any type to string
+
+type s2sTestCase struct {
```
Can we avoid all the abbreviating in these type names? I'm not sure the convenience of omitting a few characters is a good tradeoff for clarity -- code is read more than it is written, so best to keep in mind what is more convenient for the reader.

deelawn commented on co/co.go:
All throughout this file, let's not have variables named `int`. Yes, this is allowed, but it also makes reading the code slightly more confusing than it needs to be.  How about swapping out the name `int` for `point`?

deelawn reviewed (changes requested):
Thanks @tomholford -- lot's of great stuff here.  Can you make the style changes I've recommended?  All the actual content looks good, so I'll merge it once you complete those.

tomholford committed:
co: names, style, feedback

tomholford commented:
@deelawn Thank you for the prompt review. Just pushed a commit addressing the feedback.

- [x] `int` --> `point`
- [x] `Patq()` comment typo resolved
- [x] remove extraneous `patp()` comment
- [x] test helpers renamed for clarity

deelawn commented:
Awesome, thanks

deelawn commented:
I cut a v1.0.0 release that includes these changes
